File: 1776-minimum-operations-to-reduce-x-to-zero/1776-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int left=0;
        int sum=accumulate(nums.begin(),nums.end(),0);
        sum-=x;
        int current_sum=0;
        int maxlength=0;
        if(sum==0){
            return nums.size();
        }
        for(int right=0;right<nums.size();right++){
            current_sum+=nums[right];
            while(left<=right && current_sum>sum){
                current_sum-=nums[left];
                left++;
            }
            if(current_sum==sum){
                maxlength=max(maxlength,right-left+1);
            }
        }
        return (maxlength)?nums.size()-maxlength:-1;
    }
};

File: 1776-minimum-operations-to-reduce-x-to-zero/1776-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        left=0
        s=sum(nums)-x
        cs=0
        maxlen=0
        if s==0:
            return len(nums)
        for right in range(len(nums)):
            cs+=nums[right]
            while(left<=right and cs>s):
                cs-=nums[left]
                left+=1
            if(s==cs):
                maxlen=max(maxlen,right-left+1)
        return (len(nums)-maxlen) if(maxlen>0) else -1